packages/server/src/login/login-course.service.ts

```diff
@@ -86,13 +86,10 @@ export class LoginCourseService {
     return user;
   }
 
-  public async courseSectionToCourse(
-    courseName: string,
-    courseSection: number,
-  ): Promise<CourseModel> {
+  public async courseCRNToCourse(courseCRN: number): Promise<CourseModel> {
     const courseSectionModel = (
       await CourseSectionMappingModel.find({
-        where: { genericCourseName: courseName, section: courseSection },
+        where: { crn: courseCRN },
```

Comment:
This works because old courses _should_ be disabled, but that is through personal maintenance (running scripts and whatnot)

@IrisLiu-00 should we validate the semester name just to double check?

Reply:
Yeah I think this function should also take in a semester, since CRNs can potentially repeat between semesters. The MappingModel doesn't have a semester attribute but the CourseModel does - you'll have to do some joins and filters here. You'll also need to map the semester string from the admin data to our SemesterModels :P

Reply:
That makes sense, will attempt this now

Reply:
For some more information: the Admin data will probably be formatted something like `Summer_Full 21`. 

Internally, we use the full year 2021 to prevent Y21K, so you make need to do a small bit of logic on the year there to make sure you query the right semester.

Reply:
^ admin semester string format is in that notion doc outlining the admin post request

Reply:
tyty